# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: layout/generic/nsLineLayout.h
# Commit: b8a405d2e140
# Full Hash: b8a405d2e140de87b7a8b06e95a04f984a8ddf0a
# Author: Jonathan Kew <jkew@mozilla.com>
# Date: 2023-09-30 21:19:54
# Regressor Bug: 1731541
# File Overlap Count: 1
# Description:
#   Bug 1731541 - Implement text-wrap: balance for nsBlockFrame reflow. r=emilio
#   
#   A simple form of balance for short blocks, implemented by incrementally reducing
#   the effective inline-size used during line-breaking, up to the point where an
#   extra line would be created.
# ==============================================================================

diff -r 2a7646e1a4a5 -r b8a405d2e140 layout/generic/nsLineLayout.h
--- a/layout/generic/nsLineLayout.h	Sat Sep 30 15:53:11 2023 +0000
+++ b/layout/generic/nsLineLayout.h	Sat Sep 30 15:53:12 2023 +0000
@@ -48,7 +48,10 @@
   void BeginLineReflow(nscoord aICoord, nscoord aBCoord, nscoord aISize,
                        nscoord aBSize, bool aImpactedByFloats,
                        bool aIsTopOfPage, mozilla::WritingMode aWritingMode,
-                       const nsSize& aContainerSize);
+                       const nsSize& aContainerSize,
+                       // aInset is used during text-wrap:balance to reduce
+                       // the effective available space on the line.
+                       nscoord aInset = 0);
 
   void EndLineReflow();
 
@@ -494,6 +497,7 @@
     nscoord mIStart;
     nscoord mICoord;
     nscoord mIEnd;
+    nscoord mInset;
 
     nscoord mBStartLeading, mBEndLeading;
     nscoord mLogicalBSize;